content/markdown/02-working-with-data/02-02-filtering-subsetting.rmd

```diff
@@ -0,0 +1,148 @@
+
+library(lterdatasampler)
+library(tidyverse)
+```
+
+If we want to filter a dataset based on a single value in a column, we can make use of Boolean operators.
+
+```{r}
+# first call the dataset from the lterdatasampler package
+crabs <- lterdatasampler::pie_crab
+```
+
+Let's look at the columns
+```{r}
+# this will show us the first six rows
+head(crabs)
+```
+
+We see there is a column called `site`. We can get the unique values of that column:
+```{r}
+unique(crabs$site)
+```
+So we might want to filter our data such that we have a new dataframe, where `site` is **only** equal to "VCR". 
+
+### Core R
+
+Filtering in core R makes use of the square bracket indexing discussed elsewhere. Recall that the index goes *[row,column]*. In the *row* position, we use the `which()` function, which simply returns where some indices are TRUE. Since we want to keep all the columns in our dataset, we leave the *column* position empty, indicating that all columns should be retained. In our example that looks like this: 
+```{r}
+vcr_crabs <- crabs[which(crabs$site == "VCR"),]
+```
+We can check it worked by calling `unique()` on the new dataframe as above:
+```{r}
+unique(vcr_crabs$site)
+```
+And we can see it worked as expected. 
+
+### Tidyverse
+
+Filtering in the Tidyverse uses mostly the `dplyr` package, which is EXTREMELY useful for most data management exercises. Here we use the aptly named `filter()` function to perform the operation. 
+```{r}
+vc_crabs <- crabs %>% 
+  dplyr::filter(site == "VCR")
+```
+The `%>%` operator to pass the original data object `crabs` to our function, `filter()`, and the Boolean operator `==` is the only argument needed for this function. 
+
+----------------
+
+There are many permutations of this single value option with Boolean operators. For example, using either approach, we could filter asking R to select rows where `site == "VCR" | site == "BC"` which would select rows where the site was either "VCR" or "BC". 
+
+In the case where we had a separate vector dictating the focal sites we were interested in, we could use that vector in combination with the `%in%` operator to deal with this: 
+
+### Core R
+```{r}
+focal_sites <- c("ZI", "RC", "VCR", "DB", "JC")
+
+focal_crabs <- crabs[which(crabs$site %in% focal_sites),]
+```
+
+### Tidyverse
+```{r}
+focal_sites <- c("ZI", "RC", "VCR", "DB", "JC")
+
+focal_crabs <- crabs %>% 
+  dplyr::filter(site %in% focal_sites)
+```
+
+## Filter Based on Multiple Values
+
+This filtering can be extended to use multiple columns relatively simply. There is a variable in this dataset called size. 
+```{r}
+hist(crabs$size)
+```
+We may want to filter crabs only in our focal sites, but also those above some size (e.g. 15).
+
+### Core R
+```{r}
+site_size <- crabs[which(crabs$site %in% focal_sites & 
+                           crabs$size > 15), ]
+```
+
+### Tidyverse
+
+```{r}
+site_size <- crabs %>% 
+  dplyr::filter(site %in% focal_sites & size > 15)
```

Comment:
might be an opportunity to show sequential piping? you could give an example where site is filtered first and then passed to a filter for size